rust/xaynet-server/src/storage/traits.rs

```diff
@@ -207,6 +207,44 @@ where
     async fn is_ready(&mut self) -> StorageResult<()>;
 }
 
+#[async_trait]
+/// An abstract trust anchor provider.
+pub trait TrustAnchor
+where
+    Self: Clone + Send + Sync + 'static,
+{
+    /// Publishes a proof of the global model.
+    ///
+    /// # Behavior
+    ///
+    /// Return `StorageResult::Ok(())` if the proof was published successfully,
+    /// otherwise return `StorageResult::Err(error)`.
+    async fn publish_proof(&mut self, global_model: &Model) -> StorageResult<()>;
+
+    /// Checks if the [`TrustAnchor`] is ready to process requests.
+    ///
+    /// # Behavior
+    ///
+    /// If the [`TrustAnchor`] is ready to process requests, return `StorageResult::Ok(())`.
+    /// If the [`TrustAnchor`] cannot process requests because of a connection error,
+    /// for example, return `StorageResult::Err(error)`.
+    async fn is_ready(&mut self) -> StorageResult<()>;
+}
+
+#[async_trait]
+pub trait Storage: CoordinatorStorage + ModelStorage + TrustAnchor {
```

Comment:
Very neat :) Funny how such an innocent looking trait leads to such improvement through the entire state machine. Why is the `is_ready` needed though? CoordinatorStorage, ModelStorage and TrustAnchor all have an `is_ready` method already.

Reply:
@little-dude 🎉

> Very neat :) Funny how such an innocent looking trait leads to such improvement through the entire state machine.

Yeah I agree😊. It is awesome to see that it work this way without dynamic dispatch. 

The `is_ready` method [combines](https://github.com/xaynetwork/xaynet/pull/636/files#diff-8bb0449496e1ce34c29fa92d3632559dc3727887c25e4c92724a2f6f2ad45adcR198) the `is_ready` methods of `CoordinatorStorage`, `ModelStorage` and `TrustAnchor` and is called in the state machine [error phase](https://github.com/xaynetwork/xaynet/pull/636/files#diff-649bb08de88168dfeb4b393ae9f3a18134c43f7cec858e5d62927496bd155de7R63).

Reply:
@little-dude !! are you dynamically dispatching meals to people doorsteps already?